### PR TITLE
Refactored duplicated tests into a parameterized one

### PR DESCRIPTION
**Pull request by eas5**
**Problem:**
Classes containing test methods with repeated test code steps, typically, parts that set up test fixtures, generate additional effort in case of test maintenance.

**Solution:**
Parameterized tests make it possible to run a test multiple times, with different arguments, as individual and independent tests. In this refactoring, no original assertion parameter was changed. A further refactoring can also be applied as soon as JUnit 5 library version is updated to at least version 5.7, which is indicated in test class.

### CODE REVIEW

File: tests/src/test/java/IOTests.java [hunk 1]
```diff
@@ -3,29 +3,32 @@
 import mindustry.game.*;
 import mindustry.io.*;
 import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
```
Comment (Anuken) [inline tests/src/test/java/IOTests.java:6]: Use wildcard imports here ([CONTRIBUTING.MD](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md#follow-the-formatting-guidelines)).

File: tests/src/test/java/IOTests.java [hunk 2]
```diff
@@ -3,29 +3,32 @@
...
+    void writeStringTest(String string){
         ByteBuffer buffer = ByteBuffer.allocate(500);
-        TypeIO.writeString(buffer, "这个服务器可以用自己的语言说话");
+        TypeIO.writeString(buffer, string);
         buffer.position(0);
-        assertEquals(TypeIO.readString(buffer), "这个服务器可以用自己的语言说话");
+        assertEquals(TypeIO.readString(buffer), string);
     }
 
+    /**
```
Comment (Anuken) [inline tests/src/test/java/IOTests.java:27]: Any particular reason why you can't just update the libraries to 5.7 instead?

File: tests/src/test/java/IOTests.java [hunk 3]
```diff
@@ -3,29 +3,32 @@
...
-        ByteBuffer buffer = ByteBuffer.allocate(500);
-        TypeIO.writeString(buffer, "asd asd asd asd asdagagasasjakbgeah;jwrej 23424234");
-        buffer.position(0);
-        assertEquals(TypeIO.readString(buffer), "asd asd asd asd asdagagasasjakbgeah;jwrej 23424234");
-    }
-
-    @Test
-    void writeChinese(){
+    @ParameterizedTest
+    @ValueSource(strings ={
```
Comment (Anuken) [inline tests/src/test/java/IOTests.java:16]:
```suggestion
    @ValueSource(strings = {
```

**Review by Anuken — CHANGES_REQUESTED** [review]

File: tests/src/test/java/IOTests.java (hunk 1 above)
Comment (eas5) [reply tests/src/test/java/IOTests.java]: Done

File: tests/src/test/java/IOTests.java (hunk 2 above)
Comment (eas5) [reply tests/src/test/java/IOTests.java]: No particular reason. I have updated the libraries

**Pull request merged**